include/core/shape.h

```diff
@@ -936,12 +2182,185 @@ class CompositeShape : public Shape<dim>
+              constituent->get_bounding_box_half_length()[2];
+            component_bounding_box_vertex[5][0] =
+              constituent->get_bounding_box_center()[0] -
+              constituent->get_bounding_box_half_length()[0];
+            component_bounding_box_vertex[5][1] =
+              constituent->get_bounding_box_center()[1] +
+              constituent->get_bounding_box_half_length()[1];
+            component_bounding_box_vertex[5][2] =
+              constituent->get_bounding_box_center()[2] -
+              constituent->get_bounding_box_half_length()[2];
+            component_bounding_box_vertex[6][0] =
+              constituent->get_bounding_box_center()[0] -
+              constituent->get_bounding_box_half_length()[0];
+            component_bounding_box_vertex[6][1] =
+              constituent->get_bounding_box_center()[1] -
+              constituent->get_bounding_box_half_length()[1];
+            component_bounding_box_vertex[6][2] =
+              constituent->get_bounding_box_center()[2] +
+              constituent->get_bounding_box_half_length()[2];
+            component_bounding_box_vertex[7][0] =
+              constituent->get_bounding_box_center()[0] -
+              constituent->get_bounding_box_half_length()[0];
+            component_bounding_box_vertex[7][1] =
+              constituent->get_bounding_box_center()[1] -
+              constituent->get_bounding_box_half_length()[1];
+            component_bounding_box_vertex[7][2] =
+              constituent->get_bounding_box_center()[2] -
+              constituent->get_bounding_box_half_length()[2];
+            for (unsigned int i = 0; i < 8; ++i)
+              {
+                // position the vextex point in the reference frame of the
+                // composite
+                auto new_point = constituent->get_rotation_matrix() *
+                                   component_bounding_box_vertex[i] +
+                                 point_nd_to_3d(constituent->get_position());
+                if (xyz_min_max_is_initialized == false)
+                  {
+                    x_max                      = new_point[0];
+                    x_min                      = new_point[0];
+                    y_max                      = new_point[1];
+                    y_min                      = new_point[1];
+                    z_max                      = new_point[2];
+                    z_min                      = new_point[2];
+                    xyz_min_max_is_initialized = true;
+                  }
+                else
+                  {
+                    x_max = std::max(x_max, new_point[0]);
+                    x_min = std::min(x_min, new_point[0]);
+                    y_max = std::max(y_max, new_point[1]);
+                    y_min = std::min(y_min, new_point[1]);
+                    z_max = std::max(z_max, new_point[2]);
+                    z_min = std::min(z_min, new_point[2]);
```

Comment:
Why have these xmax xmin etc. varibles. Can't you have an array directly instead and put this into a loop? It would be a lot more readable.